TypeScript.YAML-tmLanguage

```diff
@@ -26,11 +26,45 @@ repository:
       '1': { name: variable.ts }
     end: (?=$|[;,])
     patterns:
+    - include: '#ternary-expression'
     - include: '#type-annotation'
     - include: '#string'
     - include: '#comment'
     - include: '#expression'
 
+  ternary-expression:
+    begin: (?=\?)
+    end: (?=$|[;,])
+    patterns:
+    - include: '#ternary-operator'
+    - include: '#ternary-expression-type' 
+
+  ternary-operator:
+    begin: (\?)
+    end: (:)
+    patterns:
+    - include: '#ternary-expression-type' 
+
+  ternary-expression-type:
```

Comment:
Can you document (in comments) exactly what this is supposed to capture? For instance, why do you need types covered here?